packages/next/src/client/image.tsx

```diff
@@ -717,19 +717,27 @@ const Image = forwardRef<HTMLImageElement | null, ImageProps>(
             throw new Error(
               `Image with src "${src}" is missing required "width" property.`
             )
-          } else if (isNaN(widthInt)) {
+          } else if (Number.isNaN(widthInt)) {
             throw new Error(
               `Image with src "${src}" has invalid "width" property. Expected a numeric value in pixels but received "${width}".`
             )
+          } else if (!Number.isFinite(widthInt)) {
```

Comment:
Lets move the check for infinity into the `getInt()` function. Then we won't need separate error messages.

Reply:
Something like this:

```ts
function getInt(x: unknown): number | undefined {
 let n: number | undefined
  if (typeof x === 'number' || typeof x === 'undefined') {
    n = x
  } else if (typeof x === 'string' && /^[0-9]+$/.test(x)) {
    n = parseInt(x, 10)
  } else {
    n = NaN
  }
  
  return Number.isFinite(n) ? n : NaN
}
```

Reply:
Thank you for the idea, I've done it, review pls.

Reply:
@li-jia-nan The tests are still necessary to prove that Infinity will now print an error

Reply:
I have added test cases, review pls.